spyder/app/mainwindow.py

```diff
@@ -2861,7 +2853,9 @@ def closing(self, cancelable=False):
         for plugin in (self.widgetlist + self.thirdparty_plugins):
             # New API
             try:
-                plugin.close_window()
+                if plugin.is_undocked():

```

Comment:
This was the problem of why all plugins were appearing. The new API needs to check that the plugin was undocked.